crosslink/internal/work.go

```diff
@@ -0,0 +1,154 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crosslink
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.uber.org/zap"
+	"golang.org/x/mod/modfile"
+)
+
+// Work is the main entry point for the work subcommand.
+func Work(rc RunConfig) error {
+	rc.Logger.Debug("Crosslink run config", zap.Any("run_config", rc))
+
+	uses, err := intraRepoUses(rc.RootPath)
+	if err != nil {
+		return fmt.Errorf("failed to find Go modules: %w", err)
+	}
+
+	goWork, err := openGoWork(rc)
+	if errors.Is(err, os.ErrNotExist) {
+		goWork = &modfile.WorkFile{
+			Syntax: &modfile.FileSyntax{},
+		}
+		if addErr := goWork.AddGoStmt(rc.GoVersion); addErr != nil {
+			return fmt.Errorf("failed to create go.work: %w", addErr)
+		}
+	} else if err != nil {
+		return err
+	}
+
+	insertUses(goWork, uses, rc)
+	pruneUses(goWork, uses, rc)
+
+	return writeGoWork(goWork, rc)
+}
+
+func intraRepoUses(rootPath string) ([]string, error) {
+	var uses []string
+	err := fs.WalkDir(os.DirFS(rootPath), ".", func(path string, dir fs.DirEntry, err error) error {
```

Comment:
This seems to have overlapping functionality with https://github.com/open-telemetry/opentelemetry-go-build-tools/blob/c0765fe1477eddc1a559c752ff755d67baf152de/crosslink/internal/graph.go#L32-L59

Should they be unified into their own function?

Reply:
PTAL https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/311/commits/5c23a925391de221b112e0c296d9b2b80a944d7f